main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -901,18 +906,30 @@ private void storeCache() {
                     new Action1<Integer>() {
                         @Override
                         public void call(final Integer selectedListId) {
-                            storeCache(selectedListId, new StoreCacheHandler(CacheDetailActivity.this, progress));
+                            if (cache.isOffline()) {
+                                // cache already offline, just add to another list
+                                DataStore.addToList(Collections.singletonList(cache), selectedListId);
+                                new StoreCacheHandler(CacheDetailActivity.this, progress).sendEmptyMessage(CancellableHandler.DONE);
+                            } else {
+                                storeCache(selectedListId, new StoreCacheHandler(CacheDetailActivity.this, progress));
+                            }
                         }
-                    }, true, StoredList.TEMPORARY_LIST.id);
+                    }, true, cache.getLists());
         } else {
-            storeCache(StoredList.TEMPORARY_LIST.id, new StoreCacheHandler(this, progress));
+            if (cache.isOffline()) {
+                // cache already offline, just add to another list
+                DataStore.addToList(Collections.singletonList(cache), StoredList.TEMPORARY_LIST.id);
+                new StoreCacheHandler(CacheDetailActivity.this, progress).sendEmptyMessage(CancellableHandler.DONE);
+            } else {
+                storeCache(StoredList.TEMPORARY_LIST.id, new StoreCacheHandler(this, progress));
+            }
         }
```

Comment:
This fragment, which is more a repetition of the fragment above, should be refactored to not be repeated.